src/xtopcom/xverifier/src/xblacklist_verifier.cpp

```diff
@@ -3,47 +3,70 @@
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 #include "xverifier/xblacklist_verifier.h"
+#include "xconfig/xpredefined_configurations.h"
+#include "xconfig/xconfig_register.h"
+#include "xverifier/xverifier_errors.h"
 
 NS_BEG2(top, xverifier)
 
 bool xtop_blacklist_utl::is_black_address(std::string const& addr) {
-    auto const& black_addrs = black_config();
-    return std::find(black_addrs.begin(), black_addrs.end(), addr) != std::end(black_addrs);
+    auto black_addrs = black_config();
+    if (!black_addrs.empty()) {
+        if (std::find(black_addrs.begin(), black_addrs.end(), addr) != std::end(black_addrs)) {
+            return true;
+        }
+    }
+    return false;
+}
+bool xtop_blacklist_utl::is_black_address(std::string const& tx_source_addr, std::string const& tx_target_addr) {
+    auto black_addrs = black_config();
```

Comment:
if ( is_black_address(tx_source_addr) || is_black_address(tx_target_addr) ){
   return true;
}
return false;

Reply:
for performance